src/main/scala/gameengine/GameEngine.scala

```diff
@@ -0,0 +1,109 @@
+package gameengine
+
+import characters.{GameBoard, Person, Player, Resources}
+
+import scala.Console.println
+import scala.collection.mutable.ListBuffer
+import scala.io.StdIn.readLine
+import scala.util.Random
+
+class GameEngine {
+  val resources = new Resources
+  private var questionList: List[String] = resources.listOfQuestions
+  private val random = new Random()
+  // Instantiate the game board
+  val gameBoard = new GameBoard(resources.charactersList)
+  // create firstPlayer
+  private val firstPlayer: Player = createPlayerAndAssignGameBoard()
+  private var continuedPlaying: Boolean = true
+
+  /**
+   * Create an object of player by assigning name, gameBoard and secret characters
+   * @return player object
+   */
+  def createPlayerAndAssignGameBoard(): Player = {
+    val player = new Player(name = "Player1", gameBoard = gameBoard.gameBoardForPlayer,
+      selectRandomCharacter(resources.charactersList))
+    player
+  }
+
+
+  /**
+   * Select random character by from Character List from Resources class
+   * @param characterList list of characters from resources class
+   * @return randomCharacter of type Person
+   */
+  def selectRandomCharacter(characterList: List[Person]): Person = {
+    val randomCharacter = characterList(random.nextInt(characterList.length))
+    randomCharacter
+  }
+
+  /**
+   * Select random question from the question list and filters out
+   * questionList which is returned so that there is no repeated questions
+   * @return if
+   *         * list is empty returns there are `no more questions`
+   *         else returns the selected random question
+   */
+  def selectRandomQuestions(): String = {
+    val questionsLength = random.nextInt(questionList.length)
+    val question = questionList(questionsLength)
+    questionList = questionList.filterNot(_ == question)
+    question
+  }
+
```

Comment:
Good idea to filter out already asked questions. As you have computer guessing it could randomly pick the same one again.

Reply:
Thanks